app/Importers/DataImporter.php

```diff
@@ -59,18 +59,19 @@ public function translate(string $word, string $language): string
 
     protected function countryNotFound(string $cityName, string $countryName): void
     {
-        CityWithoutAssignedCountry::query()
-            ->withTrashed()
-            ->updateOrCreate(
-                [
-                    "city_name" => $cityName,
-                    "country_name" => $countryName,
-                ],
-                [
-                    "city_name" => $cityName,
-                    "country_name" => $countryName,
-                ],
-            );
+        $cityAttributes = [
+            "city_name" => $cityName,
+            "country_name" => $countryName,
+        ];
+
+        $city = CityWithoutAssignedCountry::query()->withTrashed()->updateOrCreate(
+            $cityAttributes,
+            $cityAttributes,
+        );
```

Comment:
What if I need to create a city but it is deleted (destroyed). Should this city be returned to the list of cities?

Reply:
But if you want to add city which is deleted you can. There's no correlation between `City` and `CityWithoutAssignedCountry`.